### PR TITLE
[BD-6661] Add isElevated prop to BpkDividedCard to swith off card shadow

### PR DESCRIPTION
**Pull request by Supremeyh**
##  Context

We're going to introduce new card wrapper component, some [discussion](https://skyscanner.slack.com/archives/C04G18ZK2TD/p1671558413954689) here. This will need a new prop `isElevated` in the `BpkDividedCard` component to be able to switch on/off the card shadow so that it wouldn't overlap with the shadow of the card wrapper [Figma](https://www.figma.com/file/irZ3YBx8vOm16ICkAr7mB3/Backpack?node-id=22572%3A28385&t=twcgLziMhOKjFfnF-0).

## Change Description
 - Add prop `isElevated` to switch on/off the card shadow so that it wouldn't overlap with the shadow of the card wrapper.

### Screenshots

| Before                        | After                         |
| ----------------------------- | ----------------------------- |
|  |  |

Remember to include the following changes:

- [x] `UNRELEASED.md` (See [How to write a good changelog entry](https://github.com/Skyscanner/backpack/blob/main/CHANGELOG_FORMAT.md))
- [x] `README.md`
- [x] Tests
- [x] Storybook examples created/updated

### CODE REVIEW

File: packages/bpk-component-card/src/BpkDividedCard.js [hunk 1]
```diff
@@ -52,6 +54,7 @@ const BpkDividedCard = (props: Props) => {
   const classNames = getClassName(
     'bpk-divided-card',
     isVertical ? 'bpk-divided-card--vertical' : 'bpk-divided-card--horizontal',
+    !isElevated ? 'bpk-divided-card--depressed' : '',
```
Comment (jasonhuang-sky) [inline packages/bpk-component-card/src/BpkDividedCard.js:57]:
```suggestion
    !isElevated && 'bpk-divided-card--depressed',
```

File: examples/bpk-component-card/examples.js [hunk 2]
```diff
@@ -108,12 +108,25 @@ const WithHrefDividedCardExample = () => (
...
+  </div>
+);
+
 const MixedExample = () => (
   <div>
     <DefaultExample />
     <WithoutPaddingExample />
     <DefaultDividedCardExample />
     <VerticalDividedCardExample />
+    <UnElevatedDividedCardExample />
```
Comment (jasonhuang-sky) [inline examples/bpk-component-card/examples.js:129]: Seems you forgot to update the snapshots causing the visual tests to fail

File: packages/bpk-component-card/src/BpkDividedCard.module.scss [hunk 3]
```diff
@@ -59,4 +59,14 @@ $fixed-secondary-width: 216;
       }
     }
   }
+
+  &--depressed {
+    box-shadow: none;
+
+    @include bpk-hover {
```
Comment (anambl) [inline packages/bpk-component-card/src/BpkDividedCard.module.scss:66]: it seems like on exiting the hover state the shadow re-appears for a brief moment

File: packages/bpk-component-card/src/BpkDividedCard.js [hunk 4]
```diff
@@ -52,6 +54,7 @@ const BpkDividedCard = (props: Props) => {
   const classNames = getClassName(
     'bpk-divided-card',
     isVertical ? 'bpk-divided-card--vertical' : 'bpk-divided-card--horizontal',
+    !isElevated && 'bpk-divided-card--depressed',
```
Comment (anambl) [inline packages/bpk-component-card/src/BpkDividedCard.js:57]: The naming `depressed` might be confusing here. I suggest `bpk-divided-card--no-elevation` as it's consistent with the rest of the code and with the naming of the prop `isElevated`?

File: packages/bpk-component-card/src/BpkDividedCard.module.scss [hunk 5]
```diff
@@ -59,4 +59,8 @@ $fixed-secondary-width: 216;
       }
     }
   }
+
+  &--depressed {
+    box-shadow: none;
```
[the remark this answers is not in the document]
Comment (Supremeyh) [reply packages/bpk-component-card/src/BpkDividedCard.module.scss:64]: It due to shadow in pseudo element, have depressed it.

File: examples/bpk-component-card/examples.js (hunk 2 above)
Comment (Supremeyh) [reply examples/bpk-component-card/examples.js]: Thanks for your review, have update storyshots via run `npm run jest:visual-tests:update` locally.

File: packages/bpk-component-card/src/BpkDividedCard.module.scss (hunk 3 above)
Comment (Supremeyh) [reply packages/bpk-component-card/src/BpkDividedCard.module.scss]: Nice catch, it due to shadow in pseudo element, have depressed it.

File: packages/bpk-component-card/src/BpkDividedCard.js (hunk 4 above)
Comment (Supremeyh) [reply packages/bpk-component-card/src/BpkDividedCard.js]: Agree on taht, have renamed it to `--no-elevation` and `NonElevatedDividedCard`.

**Review by jasonhuang-sky — APPROVED** [review]

**Review by anambl — APPROVED** [review]
LGTM!

**Pull request merged**